examples/icmp_response.rs

```diff
@@ -0,0 +1,292 @@
+        .arg(arg!(completion_ring_size: -o --completion_ring_size <completion_ring_size> "completion ring size").required(false).default_value("64"))
+        .get_matches();
+
+    let if_name = matches.get_one::<String>("interface").unwrap().clone();
+    let chunk_size = (*matches.get_one::<String>("chunk_size").unwrap())
+        .parse::<u32>()
+        .expect("Check chunk size.");
+    let chunk_count = (*matches.get_one::<String>("chunk_count").unwrap())
+        .parse::<u32>()
+        .expect("Check chunk count.");
+    let rx_ring_size = (*matches.get_one::<String>("rx_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check rx ring size.");
+    let tx_ring_size = (*matches.get_one::<String>("tx_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check tx ring size.");
+    let filling_ring_size = (*matches.get_one::<String>("filling_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check filling ring size.");
+    let completion_ring_size = (*matches.get_one::<String>("completion_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check completion ring size.");
+
+    /* signal define to end the application */
+    let term: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
+    let result_sigint: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGINT, Arc::clone(&term));
+    let result_sigterm: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term));
+    let result_sigabrt: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGABRT, Arc::clone(&term));
+
+    if result_sigint.or(result_sigterm).or(result_sigabrt).is_err() {
+        panic!("signal is forbidden");
+    }
+
+    /* save source MAC address into variable */
+    let all_interfaces: Vec<NetworkInterface> = interfaces();
+    let mathced_interface: Option<&NetworkInterface> = all_interfaces
+        .iter()
+        .find(|element| element.name.as_str() == if_name.as_str());
+
+    let src_mac_address: MacAddr = match mathced_interface {
+        Some(target_interface) => {
+            let option: Option<MacAddr> = target_interface.mac;
+            if option.is_none() {
+                panic!("couldn't find source MAC address");
+            }
+
+            option.unwrap()
+        }
+        None => {
+            panic!("couldn't find source MAC address");
+        }
+    };
+
+    /* execute pv_open() */
+    let pv_open_option: Option<pv::Nic> = pv::pv_open(
+        &if_name,
+        chunk_size,
+        chunk_count,
+        rx_ring_size,
+        tx_ring_size,
+        filling_ring_size,
+        completion_ring_size,
+    );
+    let mut nic: pv::Nic;
+    if let Some(a) = pv_open_option {
+        nic = a;
+    } else {
+        panic!("nic allocation failed!");
+    }
```

Comment:
There are no reason to be mutable.

```suggestion
    let nic = match pv_open_option {
        Some(nic) => nic,
        _ => panic!("nic allocation failed!"),
    }
```